api/include/opentelemetry/trace/propagation/detail/context.h

```diff
@@ -9,15 +10,22 @@ namespace propagation
 namespace detail
 {
 
-inline trace::SpanContext GetCurrentSpan(const context::Context &context)
+inline nostd::shared_ptr<trace::Span> GetSpanFromContext(const context::Context &context)
```

Comment:
Sure, that's a good point.

Reply:
@seemk @maxgolov  - I have renamed these functions as `GetSpan` and `SetSpan`, however, they are now part of the `trace::propagation` namespace instead of `trace`, as these functions are specific to `propagation`. Let me know if there are any concerns.